### PR TITLE
Add global launch and configuration files

### PR DESCRIPTION
**Pull request by DFriend01**
### Description

Depends on the following PRs:
- https://github.com/UBCSailbot/local_pathfinding/pull/27
- https://github.com/UBCSailbot/boat_simulator/pull/28
- https://github.com/UBCSailbot/network_systems/pull/32

If not merged, use the following branches:
- local_pathfinding: `patrick-5546/overhaul_launch_file`
- boat_simulator: `user/dfriend01/17-overhaul-launch-files`
- network_systems: `user/dfriend01/update_launch_file_name`

This PR aims to streamline the process of running all packages while having a single source for global configuration and launching.

### To Dos
- [x] Move launch file to somewhere insides `src/`, as this is where the linters run
- [x] Move global config file from local_pathfinding to this repository
- [x] Read global config file in `main_launch.py`
- [x] Decide on local launch naming convention. `<PACKAGE>_launch.py` vs `main_launch.py`, or something else.
- [x] Add debug configuration for running global launch file

### Verification

- [x] Run global launch file: `ros2 launch $ROS_WORKSPACE/global_launch/main_launch.py`
    - [x] Launches all packages as expected and all ROS topics show up in `ros2 topic list`.
    - [x] Test mode launch argument with values `production` and `development`
    - [x] Test log_level launch argument with value `debug`
    - [x] Test debug configuration
- [x] Run package launch files: ros launch debug configuration or `ros2 launch <package> main_launch.py`

### Resources
- [Invoking children launch files from a parent launch file](https://docs.ros.org/en/humble/How-To-Guides/Launch-file-different-formats.html)
- [Using launch arguments directly in parent launch file](https://github.com/jrgnicho/collaborative-robotic-sanding/blob/3902e4f0e76bde226b18a997fd60fc30e1961212/crs_application/launch/perception.launch.py#L21)

### CODE REVIEW

**Comment by DFriend01** [comment]
#### Adding arguments
To do this, I believe that the steps are the following:
1. Edit the function `get_include_launch_descriptions()` in `main_launch.py` to accept launch arguments and include them in the launch include a description. I believe section 2 of [this link](https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Using-Substitutions.html) shows you what that looks like.
2. In each of the children launch files, make sure that the global parameters are declared with the `DeclareLaunchArgument` class and include them in the `LaunchDescription` (like what local pathfinding does right now).
3. If you want to add any more global arguments, follow the instructions on line 17 of `main_launch.py` and repeat step 2.

**Comment by DFriend01** [comment]
@patrick-5546 It was decided that the following will be done:
- Global configuration files and package-specific configuration files will be **stored** in sailbot workspace
- Both the main launch file and the children will accept two arguments
  - The path to the global config file to be used
  - The path to the package-specific config file to be used
- If launched via the main launch file, whatever the main launch file specifies will **override** the arguments in the children launch files.

image

**Review by patrick-5546 — APPROVED** [review]
Great work on this PR, I really like this flow

**Comment by patrick-5546** [comment]
After this is merged let's create a minor release

File: global_launch/main_launch.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        List[IncludeLaunchDescriptions]: The launch descriptions.
+    """
+    include_launch_descriptions = []
+    for pkg in ros_package_list:
+        pkg_main_launch = os.path.join(ROS_WORKSPACE, "src", pkg, "launch", "main_launch.py")
+        launch_description = IncludeLaunchDescription(
+            launch_description_source=PythonLaunchDescriptionSource(
+                launch_file_path=pkg_main_launch
+            ),
+            launch_arguments=global_arguments,
```
[the remark this answers is not in the document]
Comment (DFriend01) [reply global_launch/main_launch.py]: This is not necessary, you are correct.

**Pull request merged**